collector/library/pod.js

```diff
@@ -93,21 +94,17 @@ class Pod {
             }
 
             if (batch.batchSize > bytesSentFromBatch) {
-                console.log('downloading batch', batch);
-                // handle batch
-                await this.handleBatch(batch);
-
-                if (await this.getBytesSentFromThisParticularBatch({ batch }) < batch.batchSize) {
-                    console.log('could not download the entire batch', batch);
-                    return;
-                } // Avoid moving to next batch in case we could not download the entire batch
+                targetBatch = batch;
+                
+                break;
             }
         }
 
-        // In case we handled everything successfully, start again immediately
-        // We usually will wait and listen on the latest batch for quite some time.. So there
-        // is less of a point to wait a complete retryIntervalInMs before continuing in case all went wells
-        return this.checkAndProcessNewBatches();
+        if (targetBatch !== undefined) {
```

Comment:
this is not accurate - if there were batches, and we already downloaded all the bytes from the last batch - what we want to do is to GET ?follow on the current...